source/brailleTables.py

```diff
@@ -633,3 +675,50 @@ def listTables():
 # Translators: The name of a braille table displayed in the
 # braille settings dialog.
 addTable("zu-za-g2.ctb", _("Zulu grade 2"), contracted=True)
+
+# Add new first maps for the custom tables - provided in the scratchpad directory
+# and/or by addons.
+_tables = _tables.new_child()
+_tablesDirs = _tablesDirs.new_child()
+
+
+def initialize():
+	# The builtin tables were added at import time to the parent map.
+	# Now, add the custom tables to the first map.
+	import addonHandler
+	for addon in addonHandler.getRunningAddons():
+		directory = os.path.join(addon.path, "brailleTables")
+		if os.path.isdir(directory):

```

Comment:
Would it perhaps be more effective to first check the manifest instead of probing the filesystem?

Reply:
Agreed, definitely going to do that. That's a part of the change I had in mind where a manifest entry is mandatory for an table to work, with the exception of inclusions.